### PR TITLE
chore: remove redundant logging calls from some message handlers

### PR DESCRIPTION
**Pull request by natrist**
## Changes Proposed:
Remove redundant logging calls from some message handlers and handle logging through the WorldSession::Update method.

This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:
Reduces redundant network logging code

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Server/WorldSession.cpp [hunk 1]
```diff
@@ -316,6 +309,7 @@ bool WorldSession::Update(uint32 diff, PacketFilter& updater)
         ClientOpcodeHandler const* opHandle = opcodeTable[opcode];
 
         METRIC_DETAILED_TIMER("worldsession_update_opcode_time", METRIC_TAG("opcode", opHandle->Name));
+        LOG_DEBUG("network", "message id %d under READ", opcode);
```
Comment (Nefertumm) [inline src/server/game/Server/WorldSession.cpp:312]:
```suggestion
        LOG_DEBUG("network", "message id {} under READ", opcode);
```

File: src/server/game/Server/WorldSession.cpp [hunk 2]
```diff
@@ -332,8 +326,7 @@ bool WorldSession::Update(uint32 diff, PacketFilter& updater)
                         requeuePackets.push_back(packet);
                         deletePacket = false;
 
-                        LOG_DEBUG("network", "Re-enqueueing packet with opcode {} with with status STATUS_LOGGEDIN. "
-                                    "Player {} is currently not in world yet.", GetOpcodeNameForLogging(static_cast<OpcodeClient>(packet->GetOpcode())), GetPlayerInfo());
+                        LOG_DEBUG("network", "Delaying processing of message with status STATUS_LOGGEDIN: No players in the world for account id %d", GetAccountId());
```
Comment (Nefertumm) [inline src/server/game/Server/WorldSession.cpp:329]:
```suggestion
                        LOG_DEBUG("network", "Delaying processing of message with status STATUS_LOGGEDIN: No players in the world for account id {}", GetAccountId());
```

**Comment by heyitsbench** [comment]
@Kitzunu

File: src/server/game/Server/WorldSession.cpp [hunk 3]
```diff
@@ -316,6 +309,7 @@ bool WorldSession::Update(uint32 diff, PacketFilter& updater)
         ClientOpcodeHandler const* opHandle = opcodeTable[opcode];
 
         METRIC_DETAILED_TIMER("worldsession_update_opcode_time", METRIC_TAG("opcode", opHandle->Name));
+        LOG_DEBUG("network", "message id {} under READ", opcode);
```
Comment (Winfidonarleyan) [inline src/server/game/Server/WorldSession.cpp:312]: I think need use `GetOpcodeNameForLogging(static_cast<OpcodeClient>(packet->GetOpcode()))`

File: src/server/game/Server/WorldSession.cpp [hunk 4]
```diff
@@ -255,7 +249,6 @@ void WorldSession::SendPacket(WorldPacket const* packet)
         return;
     }
 
-    LOG_TRACE("network.opcode", "S->C: {} {}", GetPlayerInfo(), GetOpcodeNameForLogging(static_cast<OpcodeServer>(packet->GetOpcode())));
```
Comment (Winfidonarleyan) [inline src/server/game/Server/WorldSession.cpp:258]: I don't think it's worth deleting it

File: src/server/game/Server/WorldSession.cpp (hunk 3 above)
Comment (natrist) [reply src/server/game/Server/WorldSession.cpp:312]: The point of logging the ID is efficiency.

File: src/server/game/Server/WorldSession.cpp (hunk 4 above)
Comment (natrist) [reply src/server/game/Server/WorldSession.cpp:258]: The packet is already logged to file one call later, in [WorldSocket::SendPacket](https://github.com/azerothcore/azerothcore-wotlk/blob/062591c3c4a5978e5c18c7a4dfb2ae6ea6a5c321/src/server/game/Server/WorldSocket.cpp#L428).

File: src/server/game/Server/WorldSession.cpp (hunk 3 above)
Comment (Kitzunu) [reply src/server/game/Server/WorldSession.cpp:312]: I would prefer the actual OP code name to be sent here. For convenience

File: src/server/game/Server/WorldSession.cpp (hunk 4 above)
Comment (Kitzunu) [reply src/server/game/Server/WorldSession.cpp:258]: Good to have it sent in console too.

File: src/server/game/Server/WorldSession.cpp [hunk 5]
```diff
@@ -332,8 +326,7 @@ bool WorldSession::Update(uint32 diff, PacketFilter& updater)
                         requeuePackets.push_back(packet);
                         deletePacket = false;
 
-                        LOG_DEBUG("network", "Re-enqueueing packet with opcode {} with with status STATUS_LOGGEDIN. "
-                                    "Player {} is currently not in world yet.", GetOpcodeNameForLogging(static_cast<OpcodeClient>(packet->GetOpcode())), GetPlayerInfo());
+                        LOG_DEBUG("network", "Delaying processing of message with status STATUS_LOGGEDIN: No players in the world for account id {}", GetAccountId());
```
Comment (Kitzunu) [inline src/server/game/Server/WorldSession.cpp:329]: why are we removing the opcode name here

Comment (natrist) [reply src/server/game/Server/WorldSession.cpp:329]: It is already logged 17 lines above. First it logs the message ID, then it logs this line, saying the processing is delayed. Makes sense?

File: src/server/game/Server/WorldSession.cpp (hunk 4 above)
Comment (natrist) [reply src/server/game/Server/WorldSession.cpp:258]: > Good to have it sent in console too.

What's the point in showing it in the console when the screen is getting spammed with tens of lines per second under normal debugging condition? It's not even a relevant message to begin with...

File: src/server/game/Server/WorldSession.cpp (hunk 5 above)
Comment (Kitzunu) [reply src/server/game/Server/WorldSession.cpp:329]: aight

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**